pkg/storaged/btrfs/btrfs-tool.py

```diff
@@ -241,15 +243,34 @@ def cmd_poll(opt_mount):
     sys.stdout.flush()
 
 
+def unshare_mounts():
+    # The os.unshare function is available since Python 3.12, but we
+    # still need to support Python 3.11.
+    libc = ctypes.CDLL(None)
+    libc.unshare.argtypes = [ctypes.c_int]
+    ret = libc.unshare(2**17)
+    if ret != 0:
+        raise Exception("unshare(CLONE_NEWNS) failed")
+
+
 def cmd_do(uuid, cmd):
     debug(f"DO {uuid} {cmd}")
-    with mount_database() as db:
-        filesystems = list_filesystems()
-        for fs in filesystems.values():
-            if fs['uuid'] == uuid:
-                mp = force_mount_point(db, fs, opt_repair=True)
-                with context_chdir(mp):
-                    subprocess.check_call(cmd)
+    filesystems = list_filesystems()
+    for fs in filesystems.values():
+        if fs['uuid'] == uuid:
+            path = tempfile.mkdtemp()
+            dev = fs['devices'][0]
+            try:
+                unshare_mounts()
+                subprocess.check_call(["mount", "--make-rprivate", "/"])
+                subprocess.check_call(["mount", dev, path])
+                try:
+                    with context_chdir(path):
+                        subprocess.check_call(cmd)
+                finally:
+                    subprocess.check_call(["umount", path])
```

Comment:
OK for a first round of test runs -- but this could use a retry/sleep loop for catching more cases, and finally an `umount -l` (with a big fat error message that we can see in the test log)?

Reply:
I am hestitant re `umount -l`, the man page is pretty clear that this is only meant to be used during shutdown.

Reply:
Also, I really hope we need to worry about EBUSY with a private mount... but yeah.

Reply:
OK -- then let's not do the `-l`, but a retry loop to wait for short tasks like udev probes might be prudent?

Reply:
But, aint nobody be able to see this mount except this one process, no? If we wouldn't need to remove the tmpdir, we would not even need to call umount, I think. Just exiting the process would be enough. Just checking my understanding...

Reply:
Oh, you are right! The beauty of mount namespaces, they indeed robustify all of that. So then let's just drop that mount and check in the tests that there are no leftovers visible in the OS.